File: wallet/src/main/java/com/worldventures/wallet/analytics/FailedToAddCardAction.java
package com.worldventures.wallet.analytics;

import com.worldventures.janet.analytics.AnalyticsEvent;
import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.AdobeTracker;

@AnalyticsEvent(action = "wallet:Add a Card:Error Adding",
                trackers = AdobeTracker.TRACKER_KEY)
public class FailedToAddCardAction extends WalletAnalyticsAction {

   @Attribute("addcarderror") final String addCardError = "1";
   @Attribute("cardtype") final String cardType = "Payment";
   @Attribute("adderrortype") final String errorType;

   private FailedToAddCardAction(String errorType) {
      this.errorType = errorType;
   }

   public static FailedToAddCardAction noCardConnection() {
      return new FailedToAddCardAction("No Card Connection");
   }

   public static FailedToAddCardAction noNetworkConnection() {
      return new FailedToAddCardAction("No Network Connection");
   }

}

File: wallet/src/main/java/com/worldventures/wallet/analytics/FailedToAddCardAction.kt
package com.worldventures.wallet.analytics

import com.worldventures.core.service.analytics.AdobeTracker
import com.worldventures.core.service.analytics.Attribute
import com.worldventures.janet.analytics.AnalyticsEvent

@AnalyticsEvent(action = "wallet:Add a Card:Error Adding", trackers = arrayOf(AdobeTracker.TRACKER_KEY))
class FailedToAddCardAction private constructor(@field:Attribute("adderrortype") internal val errorType: String) : WalletAnalyticsAction() {

   @Attribute("addcarderror") internal val addCardError = "1"
   @Attribute("cardtype") internal val cardType = "Payment"

   companion object {

      fun noCardConnection(): FailedToAddCardAction {
         return FailedToAddCardAction("No Card Connection")
      }

      fun noNetworkConnection(): FailedToAddCardAction {
         return FailedToAddCardAction("No Network Connection")
      }
   }

}
